gateway/src/apicast/policy/default_credentials/default_credentials.lua

```diff
@@ -0,0 +1,98 @@
+--- Default credentials policy
+
+local tostring = tostring
+
+local policy = require('apicast.policy')
+local _M = policy.new('Default credentials policy')
+
+local new = _M.new
+
+function _M.new(config)
+  local self = new(config)
+
+  if config then
+    self.default_credentials = {
+      user_key = config.user_key,
+      app_id = config.app_id,
+      app_key = config.app_key
+    }
+  else
+    self.default_credentials = {}
+  end
+
+  return self
+end
+
+local function creds_missing(service)
```

Comment:
This should be moved to the Service module

Reply:
I think that in order to do this, we'd need to extract a `Credentials` class. Credentials are used in many parts of the code (service, backend_client, proxy, etc.), so this would not be an easy refactor. That's why I'm leaning towards leaving it for a future PR.

Reply:
I think we should try to avoid overloading the Service module. Just because it is coupled to 3scale data model and in the future we should work without it.

So I'd try to rather try to make very small independent components and Credentials class could work pretty well I think. But it should not really depend on "service".
I agree it is much bigger effort, because it should be designed in a way it will support various authentication methods like TLS, JWT, ...